internal/dkg/state_machine.go

```diff
@@ -328,14 +346,14 @@ func (d *DBState) Proposing(me *drand.Participant, terms *drand.ProposalTerms) (
 }
 
 // Proposed is used by non-leader nodes to set their own state when they receive a proposal
-func (d *DBState) Proposed(sender, me *drand.Participant, terms *drand.ProposalTerms) (*DBState, error) {
+func (d *DBState) Proposed(me *drand.Participant, terms *drand.ProposalTerms, metadata *drand.GossipMetadata) (*DBState, error) {
```

Comment:
I recommend making sure at a higher level that metadata isn't nil and that terms isn't nil.

Reply:
Metadata is already checked. Have added a check in `ValidateProposal` that the terms aren't `nil`. There's not an obvious place to do it at a higher level without duplicating stuff.